backend-shared/apis/managed-gitops/v1alpha1/gitopsdeployment_webhook.go

```diff
@@ -97,5 +97,9 @@ func (r *GitOpsDeployment) ValidateGitOpsDeployment() error {
 		}
 	}
 
+	if r.Spec.Destination.Environment == "" && r.Spec.Destination.Namespace != "" {
+		return fmt.Errorf("the environment field should not be empty when the namespace is non-empty")
```

Comment:
I think in general, for 'plain' strings, perhaps we can pull them out into constants in some other file, and it can be referenced here and by the tests.  I see other string instances elsewhere so it's not just this change.

Reply:
Yes right, I have updated the strings as const in this file as well as others too

Reply:
Looks good